Mathlib/Tactic/MinImports.lean

```diff
@@ -100,8 +144,11 @@ syntax "#min_imports in" term : command
 
 elab_rules : command
   | `(#min_imports in $cmd:command)=> do
+    -- in case `cmd` is a "nameless" `instance`, we produce its name
```

Comment:
(Can you elaborate why this is important, if that's brief?)
```suggestion
    -- In case `cmd` is a "nameless" `instance`, we produce its name.
    -- It is important that this is collected *before* adding the declaration to the environment.